### PR TITLE
Update warnings and fix errors

### PR DESCRIPTION
**Pull request by affanv14**

### CODE REVIEW

File: theano/gpuarray/corr_gemm.c [hunk 1]
```diff
@@ -417,6 +417,11 @@ PyGpuArrayObject* corrMM(PyGpuArrayObject *const bottom,
                 "GpuCorrMM images and kernel must have the same stack size\n");
         return NULL;
     }
+    if ((nFilters %% numgroups) != 0) {
+        PyErr_SetString(PyExc_ValueError,
+                "GPUCorrMM the number of filters must be divisible by the number of groups\n");
+        return NULL;
+    }
```
Comment (abergeron) [inline theano/gpuarray/corr_gemm.c]: You're dividing nChannels, not nFilters later.  Is this check ok?

Comment (affanv14) [inline theano/gpuarray/corr_gemm.c]: Yes, to explain in detail, input shape will be (batches, channels, ih,iw) and kernel shape will be (filters, channels/num_groups,kh,kw). So, in grouped convolution, each input shape will be (batches, channels/num_groups, ih, iw) and kernel shape will (_**filters/num_groups**_ , channels/num_groups, kh, kw). In other words, during the loop, the kernel is divided on nFilters dimension.

Comment (abergeron) [inline theano/gpuarray/corr_gemm.c]: ok

**Comment by abergeron** [comment]
jenkins test this please

**Comment by affanv14** [comment]
oops, It was a single % for GPUCorrMM

**Comment by affanv14** [comment]
i found a bug [here](https://github.com/Theano/Theano/blob/03cc25ea77670940172a1ea5e81a8ac9e66c7669/theano/gpuarray/dnn.py#L1106). Seems like an old one. should i fix it in this PR?

**Comment by abergeron** [comment]
Yes, why not.

**Comment by affanv14** [comment]
ready to be GPU tested?

**Comment by lamblin** [comment]
Jenkins test this please

**Pull request merged**